contracts/cw721-expiration/src/execute.rs

```diff
@@ -107,176 +123,72 @@ impl<'a> Cw721ExpirationContract<'a> {
-                deps, env, info, action,
-            )?,
-        )
-    }
-
-    pub fn set_withdraw_address(
-        &self,
-        deps: DepsMut,
-        sender: &Addr,
-        address: String,
-    ) -> Result<Response, ContractError> {
-        Ok(self
-            .base_contract
-            .set_withdraw_address(deps, sender, address)?)
-    }
-
-    pub fn remove_withdraw_address(
-        &self,
-        storage: &mut dyn Storage,
-        sender: &Addr,
-    ) -> Result<Response, ContractError> {
-        Ok(self
-            .base_contract
-            .remove_withdraw_address(storage, sender)?)
-    }
-
-    pub fn withdraw_funds(
-        &self,
-        storage: &mut dyn Storage,
-        amount: &Coin,
-    ) -> Result<Response, ContractError> {
-        Ok(self.base_contract.withdraw_funds(storage, amount)?)
-    }
-}
-
-// execute
-impl<'a> Cw721ExpirationContract<'a> {
-    fn transfer_nft(
-        &self,
-        deps: DepsMut,
-        env: Env,
-        info: MessageInfo,
-        recipient: String,
-        token_id: String,
-    ) -> Result<Response<Empty>, ContractError> {
-        self.assert_valid_nft(deps.as_ref(), &env, &token_id)?;
-        Ok(self
-            .base_contract
-            .transfer_nft(deps, env, info, recipient, token_id)?)
-    }
-
-    fn send_nft(
-        &self,
-        deps: DepsMut,
-        env: Env,
-        info: MessageInfo,
-        contract: String,
-        token_id: String,
-        msg: Binary,
-    ) -> Result<Response<Empty>, ContractError> {
-        self.assert_valid_nft(deps.as_ref(), &env, &token_id)?;
-        Ok(self
-            .base_contract
-            .send_nft(deps, env, info, contract, token_id, msg)?)
-    }
-
-    fn approve(
+    pub fn approve_include_nft_expired(
         &self,
         deps: DepsMut,
         env: Env,
         info: MessageInfo,
         spender: String,
         token_id: String,
         expires: Option<Expiration>,
-    ) -> Result<Response<Empty>, ContractError> {
-        self.assert_valid_nft(deps.as_ref(), &env, &token_id)?;
+    ) -> Result<Response<TCustomResponseMessage>, ContractError> {
+        self.assert_nft_expired(deps.as_ref(), &env, token_id.as_str())?;
         Ok(self
             .base_contract
             .approve(deps, env, info, spender, token_id, expires)?)
     }
 
-    fn revoke(
+    pub fn revoke_include_nft_expired(
         &self,
         deps: DepsMut,
         env: Env,
         info: MessageInfo,
         spender: String,
         token_id: String,
-    ) -> Result<Response<Empty>, ContractError> {
-        self.assert_valid_nft(deps.as_ref(), &env, &token_id)?;
+    ) -> Result<Response<TCustomResponseMessage>, ContractError> {
+        self.assert_nft_expired(deps.as_ref(), &env, token_id.as_str())?;
         Ok(self
             .base_contract
             .revoke(deps, env, info, spender, token_id)?)
     }
 
-    fn approve_all(
+    pub fn transfer_nft_include_nft_expired(
```

Comment:
Why are we changing the name here? What's the difference between approve_all and transfer nft include nft expired?

Reply:
`transfer_nft_include_nft_expired` has nothing to do with `approve_all`. Diff just shows previous line has now different code.

Also here: most logic has been moved to package. Like `approve_all`. This `cw721-expiration` has some specific logic, where expiration needs to be considered. Like for `ExecuteMsg::TransferNft`, expiration needs to be considered in contract, where as other msgs (e.g. `ApproveAll`) is independent from expiration - hence handled in package.